src/integrationtests/java/com/aws/greengrass/integrationtests/metrics/MetricsEmitterTest.java

```diff
@@ -0,0 +1,207 @@
+                authServiceRunning.countDown();
+            }
+        });
+        kernel.launch();
+        assertThat(authServiceRunning.await(TEST_TIME_OUT_SEC, TimeUnit.SECONDS), is(true));
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private void mockInstant(long expected) {
+        this.clockMock.ifPresent(ScopedMock::close);
+        Clock spyClock = spy(Clock.class);
+        MockedStatic<Clock> clockMock;
+        clockMock = mockStatic(Clock.class);
+        clockMock.when(Clock::systemUTC).thenReturn(spyClock);
+        when(spyClock.instant()).thenReturn(Instant.ofEpochMilli(expected));
+        this.clockMock = Optional.of(clockMock);
+    }
+
+
+    @Test
+    void GIVEN_kernelRunningWithMetricsConfig_WHEN_launched_THEN_metricsArePublished(ExtensionContext context)
+            throws DeviceConfigurationException, InterruptedException {
+        ignoreExceptionOfType(context, SdkClientException.class);
+        ignoreExceptionOfType(context, TLSAuthException.class);
+        //GIVEN
+        kernel.getContext().put(MqttClient.class, mqttClient);
+        kernel.getContext().put(DeviceConfiguration.class,
+                new DeviceConfiguration(kernel, MOCK_THING_NAME, "us-east-1", "us-east-1", "mock", "mock", "mock",
+                        "us-east-1", "mock"));
+
+        //Emit metric events
+        domainEvents.emit(new CertificateSubscriptionEvent(GetCertificateRequestOptions.CertificateType.SERVER,
+                CertificateSubscriptionEvent.SubscriptionStatus.SUCCESS));
+        domainEvents.emit(new AuthorizeClientDeviceActionEvent(AuthorizeClientDeviceActionEvent
+                .AuthorizationStatus.SUCCESS));
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent
+                .VerificationStatus.SUCCESS));
+        domainEvents.emit(new SessionCreationEvent(SessionCreationEvent.SessionCreationStatus.SUCCESS));
+        domainEvents.emit(new ServiceErrorEvent());
+
+        //WHEN
+        startNucleusWithConfig("metricsConfig.yaml");
+        int periodicAggregateIntervalSec = kernel.getContext().get(MetricsConfiguration.class).getAggregatePeriod();
+        assertEquals(AGGREGATE_INTERVAL, periodicAggregateIntervalSec);
+        assertNotNull(kernel.getContext().get(MetricsEmitter.class).getFuture(),
+                "periodic publish future is not scheduled");
+        long delay = kernel.getContext().get(MetricsEmitter.class).getFuture().getDelay(TimeUnit.SECONDS);
+        assertEquals(delay, periodicAggregateIntervalSec);
+
+        //move clock one day ahead to trigger publish
+        Instant aDayLater = Instant.now().plusSeconds(DEFAULT_PUBLISH_INTERVAL);
+        mockInstant(aDayLater.toEpochMilli());
+
+        //check for telemetry logs in ~root/telemetry
+        assertEquals(kernel.getNucleusPaths().rootPath().resolve("telemetry"),
+                TelemetryConfig.getTelemetryDirectory());
```

Comment:
this isn't useful. You aren't actually checking that any telemetry exists. Anyway, it is an implementation detail which you should not be validating. 

In this test you should just validate that the metrics you emit from CDA are emitting in an aggregate form over MQTT.